lib/app/utils/fhirpath_evaluator.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'rest-client'
+require 'json'
+require 'fhir_models'
+
+module Inferno
+  class FHIRPathEvaluator
+    # Returns a new FHIRPathEvaluator if not testing and the external evaluator is selected.
+    # @param selected_evaluator [String] either 'internal' or 'external'
+    # @param external_evaluator_url [String] the base url for the external evaluator
+    def self.new_evaluator(selected_evaluator, external_evaluator_url)
+      return nil unless ENV['RACK_ENV'] != 'test' && selected_evaluator == 'external'
+
+      Inferno::FHIRPathEvaluator.new(external_evaluator_url)
+    end
+
+    # @param fhirpath_url [String] the base url for the FHIRPath /evaluate endpoint
+    def initialize(fhirpath_url)
+      raise ArgumentError, 'FHIRPath URL is unset' if fhirpath_url.blank?
+
+      @fhirpath_url = fhirpath_url
+    end
+
+    # Evaluates the given FHIRPath expression against the given elements by posting each element
+    # to the FHIRPath wrapper.
+    # @param elements [Array]
+    # @param path [String]
+    def evaluate(elements, path)
+      elements = Array.wrap(elements)
+      return elements if path.blank?
+
+      types = elements.map { |e| e.class.to_s.split('::').last }
```

Comment:
```suggestion
      types = elements.map { |e| e.class.name.demodulize }
```
ActiveSupport to the rescue once again!